### PR TITLE
ci(codeql): Add codeQL from github

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to add codeQL scanning, which was recently introduced by github

Signed-off-by: Tam Mach <sayboras@yahoo.com>

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      # Only include this step if you are running this workflow on pull requests.
+      - run: git checkout HEAD^2
+        if: ${{ github.event_name == 'pull_request' }}
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        # Override language selection by uncommenting this and choosing your languages
+        with:
+          language: 'javascript'
```
Comment (twpayne) [inline .github/workflows/codeql.yaml:37]: This looks like it's scanning JavaScript, but this a Go project.

Comment (sayboras) [reply .github/workflows/codeql.yaml:37]: :wave: 

golangci-lint-action is with typescript, transpiled into javascript if I am not wrong :)

**Review by bombsimon — APPROVED** [review]
LGTM

File: .github/workflows/codeql.yaml (hunk 1 above)
Comment (twpayne) [reply .github/workflows/codeql.yaml:37]: /facepalm thanks.

**Pull request merged**